book/src/configuration.md

```diff
@@ -44,6 +44,7 @@ You may also specify a file to use for configuration with the `-c` or
 | `gutters` | Gutters to display: Available are `diagnostics` and `line-numbers` and `spacer`, note that `diagnostics` also includes other features like breakpoints, 1-width padding will be inserted if gutters is non-empty | `["diagnostics", "line-numbers"]` |
 | `auto-completion` | Enable automatic pop up of auto-completion. | `true` |
 | `auto-format` | Enable automatic formatting on save. | `true` |
+| `auto-save` | Enable automatic saving on focus moving away from Helix. | `true` |
```

Comment:
Other discussion: https://github.com/helix-editor/helix/commit/c84f3e539eedfa3f72b4ea948cc21cad942e634d#r80351369

Yes, please default this to `false`. Auto-save is contentious. I know people that love it. Personally I don't like it, I think it violates user intentions around file operations and having unwritten changes in a buffer is useful. My opinion is not so much important as recognizing that people are opinionated about this.

Precedent says that we disable contentious features by default. In recent memory: visible whitespace, indent-guides, color modes, cursorline, rulers.

> This feature isn't discoverable without reading the docs otherwise.

This comes up with other features and I sympathize with the premise but overall I disagree, especially in this case.

* A policy of enabling new features by default leads to a bloated experience.
* Without https://github.com/helix-editor/helix/issues/1633 this is not very discoverable when enabled by default. You need to notice that the modification indicator changed when switching focus and know what the modification indicator means.

Reply:
I can think of three use cases where auto-save is not optimal:

1) when you're editing a file over NFS, especially when it's over a high-latency / low bandwidth connection
2) when you've got a workflow that reloads / rebuilds an app on file change (think python or autobuild scripts)
3) when you're editing on a write-limited medium (flash or WORM)

I have no doubt that some people will find this feature useful, so I'm not arguing against its inclusion. I'm simply requesting that it not be the default, since no auto-save is the "option of least surprise" - I would suggest that very few users will expect it to be on by default, since I can't think of any other editor that enables it in a default config.

Reply:
Fair enough. The combination of being unlike other editors, hard to notice, **and** causing changes outside the editor on the filesystem are enough together to make this scary. Switched to off by default.

> 2. when you've got a workflow that reloads / rebuilds an app on file change (think python or autobuild scripts)

This is actually one of my favorite use cases for this: running a server that autoreloads on change for a webapp that autoreloads on change. With that, I can switch windows from my editor to my browser and see what I did without having to hit any other keys.

Reply:
> This is actually one of my favorite use cases for this: running a server that autoreloads on change for a webapp that autoreloads on change. With that, I can switch windows from my editor to my browser and see what I did without having to hit any other keys.

Except this breaks for things like werkzeug because if you switch away on a file that can't be interpreted, the entire server crashes and you have to restart it.